src/main/java/cloud/fogbow/rcs/core/intercomponent/RemoteFacade.java

```diff
@@ -25,16 +20,19 @@ public static RemoteFacade getInstance() {
         }
     }
     
-    public String requestService(String senderId, ServiceType serviceType) {
-        return this.factory.makeCatalogService().requestService(senderId, serviceType);
+    public HttpResponse requestService(String senderId, ServiceType serviceType) {
+        return this.catalogService.requestService(senderId, serviceType);
     }
 
     public List<ServiceType> getServices() {
-        return this.factory.makeCatalogService().getServices();
+        return this.catalogService.getServices();
     }
 
     public void cacheSave(String key, String content) {
-        this.factory.makeCatalogService().cacheSave(key, content);
+        this.catalogService.cacheSave(key, content);
+    }
+
+    public synchronized void setCatalogService(CatalogService catalogService) {
```

Comment:
why synchronized? It is only called by Main on the startup process, right?

Reply:
i'll remove this synchronized for now and in case we really see fit later, we put it back on.